consts.go

```diff
@@ -60,9 +60,11 @@ const (
 	validDataCmdRegexPattern           = `\A(?i)data\z`
 	validRsetCmdRegexPattern           = `\A(?i)rset\z`
 	validQuitCmdRegexPattern           = `\A(?i)quit\z`
-	validHeloComplexCmdRegexPattern    = `\A(` + validHeloCmdsRegexPattern + `) (` + domainRegexPattern + `|localhost)\z`
+	validHeloComplexCmdRegexPattern    = `\A(` + validHeloCmdsRegexPattern + `) (` + domainRegexPattern + `|localhost|` + addressLiteralRegexPattern + `|` + ipAddressRegexPattern + `)\z`
```

Comment:
Let's reduce interpolation items:
```go
validHeloComplexCmdRegexPattern    = `\A(` + validHeloCmdsRegexPattern + `) (` + domainRegexPattern + `|localhost|` + ipAddressRegexPattern + addressLiteralRegexPattern + `)\z`
```

Reply:
@bestwebua I am not sure what you require here. I might have misunderstood
If I remove the | and swap the patterns (like in your example above) it no longer works.
To me it will just concatenate the 2 regex's into one that will never match.

Reply:
Sorry, @dandare100, maybe I was unclear...

Your:
```go
validHeloComplexCmdRegexPattern    = `\A(` + validHeloCmdsRegexPattern + `) (` + domainRegexPattern + `|localhost|` + addressLiteralRegexPattern + `|` + ipAddressRegexPattern + `)\z`
```

I have proposed to join the pipe and addressLiteralRegexPattern as one item:
```go
validHeloComplexCmdRegexPattern    = `\A(` + validHeloCmdsRegexPattern + `) (` + domainRegexPattern + `|localhost|` + ipAddressRegexPattern + addressLiteralRegexPattern + `)\z`
```

Reply:
@bestwebua I see now. Thank you for you patience. 

Just to be sure

addressLiteralRegexPattern         = `\[` + ipAddressRegexPattern + `\]`

becomes

addressLiteralRegexPattern         = `|\[` + ipAddressRegexPattern + `\]`

Reply:
@dandare100 Yup, as I here commented before: https://github.com/mocktools/go-smtp-mock/pull/116#discussion_r966334674

Reply:
Apologies, I must open my eyes.

Will complete it as requested